src/ipc/ipc-zephyr.c

```diff
@@ -133,13 +133,24 @@ enum task_state ipc_platform_do_cmd(struct ipc *ipc)
 	/* perform command */
 	ipc_cmd(hdr);
 
-#if CONFIG_SOC_SERIES_INTEL_ADSP_CAVS
-	/* are we about to enter D3 ? */
-	if (ipc->pm_prepare_D3) {
-		/* no return - memory will be powered off and IPC sent */
+	if (ipc->task_mask & IPC_TASK_POWERDOWN) {
+#if defined(CONFIG_PM_POLICY_CUSTOM)
+		/**
+		 * @note For primary core this function
+		 * will only force set lower power state
+		 * in power management settings.
+		 * Core will enter D3 state after exit
+		 * IPC thread during idle.
+		 */
+		cpu_disable_core(PLATFORM_PRIMARY_CORE_ID);
+#else
+		/**
```

Comment:
I'd have kept the "if (ipc->pm_prepare_D3) {" check. It seems this is good as only platforms that already set TASK_POWERDOWN are using ipc-zephyr.c . Let's see what the CI thinks.